src/edu/washington/escience/myria/expression/evaluate/GenericEvaluator.java

```diff
@@ -121,27 +131,93 @@ public String getJavaExpressionWithAppend() {
   }
 
   /**
+   * Holder class for results and result counts from {@link #evaluateColumn}.
+   */
+  public static class EvaluatorResult {
+    private final ReadableColumn results;
+    private final Column<?> resultCounts;
+
+    protected EvaluatorResult(
+        @Nonnull final ReadableColumn results, @Nonnull final Column<?> resultCounts) {
+      this.results = results;
+      this.resultCounts = resultCounts;
+    }
+    /**
+     * @return a {@link ReadableColumn} containing results from {@link #evaluateColumn}
+     */
+    public ReadableColumn getResults() {
+      return results;
+    }
+    /**
+     * This is useful for, e.g., constructing a {@link TupleBatch} from several evaluators.
+     * This will return null if the result was not generated from a single-valued expression.
+     *
+     * @return a {@link Column<?>} containing results from {@link #evaluateColumn}
+     */
+    public Column<?> getResultsAsColumn() {
+      if (results instanceof Column<?>) {
```

Comment:
This is not needed anymore If we always return `Column`, see the comment in `GenericEvaluator.evaluateColumn()`.